src/main/scala/uk/gov/hmrc/play/json/Union.scala

```diff
@@ -19,15 +19,15 @@ package uk.gov.hmrc.play.json
 import play.api.libs.json._
 import scala.reflect.ClassTag
 
-class Union[A](typeField: String, readWith: PartialFunction[String, JsValue => JsResult[A]], writeWith: PartialFunction[A, JsValue]) {
+class Union[A](typeField: String, readWith: PartialFunction[String, JsValue => JsResult[A]], writeWith: PartialFunction[A, JsObject]) {
 
-  def and[B](typeTag: String)(implicit ev: B <:< A, ct: ClassTag[B], f: Format[B]) = {
+  def and[B <: A](typeTag: String)(implicit ct: ClassTag[B], f: OFormat[B]) = {
     val readCase: PartialFunction[String, JsValue => JsResult[A]] = {
       case `typeTag` => jsValue: JsValue => Json.fromJson(jsValue)(f).asInstanceOf[JsResult[A]]
     }
 
-    val writeCase: PartialFunction[A, JsValue] = {
-      case value: B => Json.toJson(value)(f).as[JsObject] ++ JsObject(Seq(typeField -> JsString(typeTag)))
```

Comment:
This code already relies on the `Format` for `B` producing `JsObject` so I have changed it to require `OFormat` instead, which is a format that always produces a `JsObject`.